### DISCUSSION TITLE
Shopping List Features

### DISCUSSION THREAD

**Discussion by nickian** [mealie-recipes/mealie#3702] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

The shopping list feature seems essential. It's currently lacking many features and is not very usable.

### Please provide a concise description of the feature that would resolve your issue.

Features that I would love to see:

- Rename shopping lists
- Duplicate shopping lists
- Allow tags for the overall list, not just items. This way you could tag a store name, for example, or event.
- Ability to archive a list once you're done with it (remove from the main screen without deleting it altogether)
- When all items are checked on a list, the list should be archived or marked complete
- Sorting options for lists and items (name, date, tag, asc/desc, etc)
- Show the date created and completed for a list
- Add shopping lists and related actions to the timeline
- Bulk add items into a list in a textarea field, one per line
- Add a field for money spend for a list

### Please consider and list out some caveats or tradeoffs made in your design decision

I don't see any tradeoffs necessary.

I don't code in Python or Typescript, but I'm happy to help with UI/UX design.

### Additional Information

- [X] If this is accepted I'm willing to submit a PR to provide this feature
- [X] If this is accepted I'm willing to help maintain this feature
- [X] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
Love these ideas! Question on this one:
> Add shopping lists and related actions to the timeline

Could you elaborate on the idea here? Is the idea to add a timeline event for when the recipe gets added to the shopping list?

And just a side note on this one:
> Add a field for money spend for a list

[Shopping lists support extras](https://docs.mealie.io/documentation/getting-started/api-usage/#shopping-list-and-food-extras), though currently they're API exclusive. I could see a use-case to expose these to the frontend, similar to how recipe API extras work.

**Reply by nickian** [reply]
As for the timeline, yes, I just meant maybe add an entry for when a shopping list is created when it is completed.

Oh, that's great. I didn't realize the API had that feature. That'd definitely be nice to have that on the frontend as well.

Thanks!

**Comment by bpmartin20** [comment]
I'm definitely interested in being able to rename a list. I name lists by their shopping date, but I'd like to be able to create a shopping list called "Next" that collects items for the next shopping trip, whenever that may be. Then when I get to actually planning meals for the next trip I'll rename it to the shopping list date. As soon after that trip that we discover something we need I'll create new "Next" to use until we know when our next trip is.

**Reply by YibingZhang325** [reply]
+1 for this feature.

I really like your “Next” workflow. I recently started using Mealie and ran into the same limitation. I created a shopping list with a temporary name while adding items, but later realized there doesn’t seem to be any way to rename it from the UI.

My only option was to create a new shopping list with the correct name and manually move everything over, which is a bit inconvenient for such a common workflow.

I think having a simple Rename option in the shopping list menu (or allowing the title to be edited directly) would make shopping lists much easier to manage.

Thanks for all the work on Mealie!
